feat: add new `filterPeaksRanges` function

jorainer opened the pull request:
- Add a new `filterPeaksRanges` function that allows to filter mass peaks in a `Spectra` by any (combination of) numeric spectra and peaks variables.
- Re-organize documentation of filter and subset operations (related to issue #288).

jorainer commented:
A nice use case for this function would be to filter a LC-MS data set keeping (or removing) all mass peaks that were considered a *chromatographic peak* by *xcms*-based preprocessing. This allows e.g. to evaluate the *foreground* (i.e. all quantified ion signal) and *background* signal (i.e. background noise, or signal that was not considered a valid chromatographic peak by *xcms*) of a full data set.

I will later add such a use case to the [xcmsTutorials](https://github.com/jorainer/xcmsTutorials) tutorial/workshop.

philouail commented on R/Spectra-functions.R:
```diff
@@ -1182,3 +1182,184 @@ processingChunkFactor <- function(x) {
+#' has to be an available spectra or peaks variable. `<range>` can be a
+#' `numeric` of length 2 defining the lower and upper boundary, or a `numeric`
+#' two-column matrix (multi-row matrices are also supported, see further
+#' below). `filterPeaksRanges(s, mz = c(200, 300))` woudl for example reduce
```
would*

philouail commented on R/Spectra-functions.R:
```diff
@@ -1182,3 +1182,184 @@ processingChunkFactor <- function(x) {
+#' the peaks matrices of the `Spectra` object `s` to mass peaks with an m/z
+#' value between 200 and 300. `filterPeaksRanges()` returns the original
+#' `Spectra` object with the filter operation added to the processing queue.
+#' Thus, the filter gets only applied when the peaks data gets extracted
```
maybe "**only**"  in bold to highlight it

philouail commented on R/Spectra-functions.R:
```diff
@@ -1182,3 +1182,184 @@ processingChunkFactor <- function(x) {
+#' or peaks variables) have to match.
+#'
+#' **Missing value handling**: any comparison which involves a missing value
+#' (beingvit a spectra variable value, a peaks variable value or a value
```
being*

philouail commented on R/Spectra-functions.R:
```diff
@@ -1182,3 +1182,184 @@ processingChunkFactor <- function(x) {
+#'
+#' **Missing value handling**: any comparison which involves a missing value
+#' (beingvit a spectra variable value, a peaks variable value or a value
+#' in one of thevprovided ranges) is treated as a logical `FALSE`. For
```
*the provided

philouail commented on R/Spectra-functions.R:
```diff
@@ -1182,3 +1182,184 @@ processingChunkFactor <- function(x) {
+#' @note
+#'
+#' In contrast to other *filter* functions, this function does not provide a
+#' `msLevel.` parameter to apply the filter only on spectra of the specified
```
extra point at `msLevel`

philouail commented on R/Spectra-functions.R:
```diff
@@ -1182,3 +1182,184 @@ processingChunkFactor <- function(x) {
+#'
+#' In contrast to other *filter* functions, this function does not provide a
+#' `msLevel.` parameter to apply the filter only on spectra of the specified
+#' MS levels. In contrast, to apply no, or different, filters to spectra from
```
mmm i'm confused by this sentence

philouail commented on R/Spectra.R:
```diff
@@ -331,31 +331,23 @@ NULL
+#' operations that reduce the number of spectra in the object as well as filters
+#' that reduce the *content* of the `Spectra` object. See section
+#' *Filter peaks data* below for functions that filter the peaks data of a
+#' `Spectra`.
```
maybe to make it extremely clear, we can say the number of spectra in the Spectra object will not change with filters until applyProcessing() is run ?

philouail replied:
I guess here or at the beginning of the filter section

philouail commented on vignettes/Spectra.Rmd:
```diff
@@ -360,25 +360,39 @@ Similar to spectra variables it is also possible to replace values for
+
+### Filter spectra data
+
+These functions comprise subset operations that reduce the total number of
```
Maybe add "directly" or immediately" ?

philouail commented on vignettes/Spectra.Rmd:
```diff
@@ -390,53 +404,59 @@ affecting the total number of spectra are (in alphabetic order):
+### Filter or aggregate mass peak data
+
+These function filter or aggregate the mass peak data (`peaksData()`) of each
+spectrum in a `Spectra` without changing the total number of spectra.
```
as before I would precise "until the applyProcessing() function in run"

philouail reviewed (approved):
The codes looks all good to me. 

Thanks for reorganizing the documentation I believe it is much more clear now. I put some comments to signals typos and maybe some suggestions. But i think it's great !

jorainer commented:
Thanks @philouail ! I addressed all of your points! Hope documentation is now a bit clearer. Happy for additional/extra feedback, otherwise I proceed and merge tomorrow.
